modules/@apostrophecms/user/lib/password-hash.js

```diff
@@ -0,0 +1,122 @@
+// Password hashing based on scrypt, per:
+// https://cheatsheetseries.owasp.org/cheatsheets/Password_Storage_Cheat_Sheet.html
+//
+// Also includes legacy support for pbkdf2 passwords.
+//
+// Adapted from the "credential" and "credentials" modules,
+// which were also released under the MIT license.
+
+const util = require('util');
+const crypto = require('crypto');
+
+const scrypt = util.promisify(crypto.scrypt);
+const pbkdf2 = util.promisify(crypto.pbkdf2);
+const randomBytes = util.promisify(crypto.randomBytes);
+const timingSafeEqual = crypto.timingSafeEqual;
+
+function getScryptOptions(options) {
+  const result = {
+    cost: 131072,
```

Comment:
From the OWASP guidelines linked above. This is significantly harder to crack than the Node.js defaults of `scrypt`